app/controllers/asset/AddAssetsController.scala

```diff
@@ -50,42 +50,45 @@ class AddAssetsController @Inject()(
                                      checkAnswersFormatters: CheckAnswersFormatters
                                    )(implicit ec: ExecutionContext) extends AddAssetController {
 
-  private val addAnotherForm: Form[AddAssets] = addAnotherFormProvider()
+  private def addAnotherForm(isTaxable: Boolean): Form[AddAssets] = addAnotherFormProvider.withPrefix(prefix(isTaxable))
   private val yesNoForm: Form[Boolean] = yesNoFormProvider.withPrefix("addAnAssetYesNo")
 
   private def actions(draftId: String): ActionBuilder[RegistrationDataRequest, AnyContent] =
     identify andThen getData(draftId) andThen requireData
 
+  private def prefix(isTaxable: Boolean) = "addAssets" + (if (isTaxable) "" else ".nonTaxable")
```

Comment:
can this be if(!isTaxable) ".nonTaxable" ?

Reply:
@jassalrichy it can yes quite right

Reply:
@jassalrichy ah no it can't haha if the `else` isn't defined it adds Unit on to the end of the string (hence the failing tests). Have reverted back but flipped the if statement to make it read a but better